sql/patches/patch41-requestform-usage.sql

```diff
@@ -0,0 +1,72 @@
+# noinspection SqlResolveForFile
+
+DROP PROCEDURE IF EXISTS SCHEMA_UPGRADE_SCRIPT;
+DELIMITER ';;'
+CREATE PROCEDURE SCHEMA_UPGRADE_SCRIPT() BEGIN
+    -- -------------------------------------------------------------------------
+    -- Developers - set the number of the schema patch here!
+    -- -------------------------------------------------------------------------
+    DECLARE patchversion INT DEFAULT 41;
+    -- -------------------------------------------------------------------------
+    -- working variables
+    DECLARE currentschemaversion INT DEFAULT 0;
+    DECLARE lastversion INT;
+	
+    -- check the schema has a version table
+    IF NOT EXISTS (SELECT * FROM information_schema.tables WHERE table_name = 'schemaversion' AND table_schema = DATABASE()) THEN
+        SIGNAL SQLSTATE '45000' SET message_text = 'Please ensure patches are run in order! This database does not have a schemaversion table.';
+    END IF;
+    
+    -- get the current version
+    SELECT version INTO currentschemaversion FROM schemaversion;
+    
+    -- check schema is not ahead of this patch
+    IF currentschemaversion >= patchversion THEN
+        SIGNAL SQLSTATE '45000' SET message_text = 'This patch has already been applied!';
+    END IF;
+    
+    -- check schema is up-to-date
+    SET lastversion = patchversion - 1;
+    IF currentschemaversion != lastversion THEN
+		SET @message_text = CONCAT('Please ensure patches are run in order! This patch upgrades to version ', patchversion, ', but the database is not version ', lastversion);
+        SIGNAL SQLSTATE '45000' SET message_text = @message_text;
+    END IF;
+
+    -- -------------------------------------------------------------------------
+    -- Developers - put your upgrade statements here!
+    -- -------------------------------------------------------------------------
+
+    alter table request
+        add originform int(10) unsigned null,
+        add constraint request_requestform_id_fk
+            foreign key (originform) references requestform (id) ;
+
+    alter table requestform
+        add usernamehelp text not null,
+        add emailhelp text not null,
+        add commentshelp text not null;
+
+    INSERT INTO requestform (updateversion, enabled, domain, name, publicendpoint, formcontent, overridequeue, usernamehelp, emailhelp, commentshelp) VALUES (9, 1, 1, 'Default form', 'default', '## Request an account!
```

Comment:
Out of curiosity, why `updateversion` of 9?

Reply:
I actually have no idea what I was thinking four months ago!

I *think* what happened was I imported the existing template, made tweaks to it through the GUI to convert from HTML to markdown, and then exported the row as a SQL insert into the migration script, and it just happened to be version 9.

This makes no appreciable difference though, updateversion is an unsigned 32-bit integer so as long as we don't make over 4 billion(ish) changes to this request form, a difference of 9 isn't going to matter :grinning: